pkg/labels/selector_test.go

```diff
@@ -306,65 +493,89 @@ func TestRequirementLabelSelectorMatching(t *testing.T) {
 func TestSetSelectorParser(t *testing.T) {
 	setSelectorParserTests := []struct {
 		In    string
-		Out   SetBasedSelector
+		Out   Selector
 		Match bool
 		Valid bool
 	}{
 		{"", &LabelSelector{Requirements: nil}, true, true},
 		{"\rx", &LabelSelector{Requirements: []Requirement{
-			getRequirement("x", Exists, nil, t),
+			getRequirement("x", Exists, nil, SetBasedMatch, t),
+		}}, true, true},
+		{"this-is-a-dns.domain.com/key-with-dash", &LabelSelector{Requirements: []Requirement{
```

Comment:
We should also cover this case in the fuller expression evaluator, e.g. "this-is-a-dns.domain.com/key-with-dash in (so,what)".

We should also have coverage for keys that look like numbers (which are actually valid).  A key called "0" is legit, I think.